src/components/logo/logo.tsx

```diff
@@ -64,15 +64,30 @@ class Logo extends Component<any, any> {
     const url = this.props.config.url;
     if (url && url.indexOf(ENTRY_VAR) !== -1) {
       const {player, eventManager} = this.props;
-      eventManager.listen(player, player.Event.MEDIA_LOADED, () => {
-        const entryId = player.sources.id;
-        if (typeof entryId === 'string') {
-          this.setState({urlLink: url.replace(ENTRY_VAR, entryId)});
-        } else {
-          this.props.logger.debug(`Logo url was not replaced; entry id was not found.`);
-          this.setState({isUrlClickable: false});
-        }
-      });
+      if (!this._setLogoUrlWithEntryId(url)) {
```

Comment:
Why Media_Loaded is conditionally listened
What about change media?
Change media is not limited to one during player lifetime

Reply:
the logic is about adding the listener only if the entry id doesn't exist yet. there are use-cases where the entry id might exist already, before the component was mounted (like in this use-case with ima).